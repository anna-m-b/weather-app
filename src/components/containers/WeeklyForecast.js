import React from 'react';
import PropTypes from 'prop-types';
import DailyForecast from '../components/DailyForecast';
import getStringyDate from '../../util/getStringyDate';

import '../../styles/WeeklyForecast.css';

const WeeklyForecast = ({ forecast }) => {
  const forecasts = forecast.map((forecastItem) => {
    const { date, temperature, description, icon } = forecastItem;
    const dateString = getStringyDate(date);
    return (
      <DailyForecast
        date={dateString}
        temperature={temperature.max}
        description={description}
        icon={icon}
        key={date}
      />
    );
  });

  return <div className="WeeklyForecast">{forecasts}</div>;
};

WeeklyForecast.propTypes = {
  forecast: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }),
      wind: PropTypes.shape({
        speed: PropTypes.number,
        direction: PropTypes.string,
      }),
      humidity: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.string,
    })
  ).isRequired,
};

export default WeeklyForecast;
