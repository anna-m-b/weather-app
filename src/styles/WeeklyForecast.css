.WeeklyForecast {
  display: flex;
  flex-direction: column;
  flex-wrap:wrap;
  align-items: center;
  border-top: 1px solid grey;
  border-bottom: 1px solid grey;
}