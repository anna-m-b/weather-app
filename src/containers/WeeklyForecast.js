import React from 'react';
import PropTypes from 'prop-types';
import DailyForecast from '../components/DailyForecast';
import '../styles/WeeklyForecast.css';

const WeeklyForecast = ({ forecasts, handleMoreDetailsClick, errorCode }) => {
  let forecastsToDisplay;
  if (errorCode) {
    forecastsToDisplay = forecasts.map((forecastItem) => {
      const { date } = forecastItem;

      return (
        <DailyForecast
          date={date}
          temperature="- - -"
          description="- - -"
          icon="800"
          key={date}
          handleMoreDetailsClick={() => {}}
        />
      );
    });
  } else if (forecasts) {
    forecastsToDisplay = forecasts.map((forecastItem) => {
      const { date, temperature, description, icon } = forecastItem;

      return (
        <DailyForecast
          date={date}
          temperature={temperature.max}
          description={description}
          icon={icon.toString()}
          key={date}
          handleMoreDetailsClick={handleMoreDetailsClick}
        />
      );
    });
  }
  return <div className="WeeklyForecast">{forecastsToDisplay}</div>;
};

WeeklyForecast.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }),
      wind: PropTypes.shape({
        speed: PropTypes.number,
        direction: PropTypes.string,
      }),
      humidity: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.number,
    })
  ).isRequired,
  handleMoreDetailsClick: PropTypes.func.isRequired,
  errorCode: PropTypes.number,
};

WeeklyForecast.defaultProps = {
  errorCode: 0,
};

export default WeeklyForecast;
