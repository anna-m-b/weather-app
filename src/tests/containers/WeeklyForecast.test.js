import React from 'react';
import { render } from '@testing-library/react';
import WeeklyForecast from '../../containers/WeeklyForecast';

// What could be the most important thing to test here? Please remember that tests for particular values already have been tested.
// What should be tested first?
// What type of queries and assertions you might want use this time?

describe('WeeklyForecast', () => {
  const validProps = [
    {
      date: 1111111,
      description: 'Stub description 1',
      icon: 'stubIcon1',
      temperature: {
        max: 22,
        min: 12,
      },
    },
    {
      date: 2222222,
      description: 'Stub description2',
      icon: 'stubIcon2',
      temperature: {
        max: 24,
        min: 13,
      },
    },
  ];

  it('renders correctly', () => {
    const { asFragment } = render(<WeeklyForecast forecasts={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the correct number of daily forecasts', () => {
    const { getAllByTestId } = render(
      <WeeklyForecast forecasts={validProps} />
    );

    expect(getAllByTestId('daily-forecast')).toHaveLength(2);
  });
});
