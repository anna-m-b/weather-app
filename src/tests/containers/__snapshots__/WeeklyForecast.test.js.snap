// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyForecast renders correctly 1`] = `
<DocumentFragment>
  <div
    class="WeeklyForecast"
  >
    <div
      class="DailyForecast"
      data-testid="daily-forecast"
    >
      <div
        class="DailyForecast_date"
      >
        Thu, 1 Jan
      </div>
      <figure
        aria-label="Stub description 1 icon"
        class="DailyForecast_icon"
        role="figure"
      >
        <i
          class="stubIcon1"
        />
      </figure>
      <div
        class="DailyForecast_temperature"
      >
        22°c
      </div>
      <div
        class="DailyForecast_description"
      >
        Stub description 1
      </div>
      <div
        class="DailyForecast_more-details"
      >
        More details
      </div>
    </div>
    <div
      class="DailyForecast"
      data-testid="daily-forecast"
    >
      <div
        class="DailyForecast_date"
      >
        Thu, 1 Jan
      </div>
      <figure
        aria-label="Stub description2 icon"
        class="DailyForecast_icon"
        role="figure"
      >
        <i
          class="stubIcon2"
        />
      </figure>
      <div
        class="DailyForecast_temperature"
      >
        24°c
      </div>
      <div
        class="DailyForecast_description"
      >
        Stub description2
      </div>
      <div
        class="DailyForecast_more-details"
      >
        More details
      </div>
    </div>
  </div>
</DocumentFragment>
`;
